src/Node/Buffer/Mutable.purs

```diff
@@ -0,0 +1,228 @@
+module Node.Buffer.Mutable
+ ( class MutableBuffer
+ , create
+ , freeze
+ , thaw
+ , fromArray
+ , fromString
+ , fromArrayBuffer
+ , toArrayBuffer
+ , read
+ , readString
+ , toString
+ , write
+ , writeString
+ , toArray
+ , getAtOffset
+ , setAtOffset
+ , size
+ , concat
+ , concat'
+ , copy
+ , fill
+ , EffectBuffer
+ , STBuffer
+ , runST
+ ) where
+
+import Prelude
+
+import Control.Monad.ST (ST, kind Region)
+import Control.Monad.ST as ST
+import Data.ArrayBuffer.Types (ArrayBuffer)
+import Data.Maybe (Maybe)
+import Effect (Effect)
+import Node.Buffer (Buffer, BufferValueType, Octet, Offset)
+import Node.Buffer as Buffer
+import Node.Encoding (Encoding, encodingToNode)
+import Unsafe.Coerce (unsafeCoerce)
+
+-- | A type class for mutable buffers `b` where operations on those buffers are
+-- | represented by a particular effect type `e`.
+class MutableBuffer b e | e -> b, b -> e where
```

Comment:
I have no objections to this. I didn't add a `Monad` constraint before because it was not strictly needed. I suppose having the constraint means you can write generic code more easily - since you only need the `MutableBuffer` constraint on your code and not `Monad` too. I am curious if this is the only motivating reason to add the constraint? (I have added it anyway, in https://github.com/purescript-node/purescript-node-buffer/pull/33/commits/9b48b294faee45bfee4e1372ecb14e1d22527dcd)

Reply:
Indeed, but that's not the only reason - if we want to add laws, which I think is quite possible at some point (although probably isn't necessary right away) - we'll need the Monad constraint to be able to express them.

Reply:
Okay, fair enough.